runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/AbstractErrorTypesValidation.java

```diff
@@ -73,17 +63,18 @@ public Level getLevel() {
         : WARN;
   }
 
-  protected static boolean errorMappingPresent(ComponentAst operationComponent) {
-    if (!operationComponent.getModel(ParameterizedModel.class).isPresent()) {
+  protected boolean isErrorTypePresent(ComponentAst component) {
+    if (getErrorTypeParam(component) == null) {
       return false;
     }
-    final ComponentParameterAst errorMappingsAst =
-        operationComponent.getParameter(ERROR_MAPPINGS, ERROR_MAPPINGS_PARAMETER_NAME);
-    return errorMappingsAst != null && errorMappingsAst.getValue().getValue().isPresent();
-  }
 
-  protected static List<ErrorMapping> getErrorMappings(ComponentAst component) {
-    return (List<ErrorMapping>) component.getParameter(ERROR_MAPPINGS, ERROR_MAPPINGS_PARAMETER_NAME).getValue().getRight();
+    if (waiveUnresolvedPropertiesOnParams) {
+      if (getErrorTypeParam(component).getRawValue().contains("${")) {
```

Comment:
This logic should belong to some api, and be more correct. Please add it or create a ticket and add a TODO